### PR TITLE
Bug 1813221: jsonnet/telemeter: set cpu requests, unset limits

### PR DESCRIPTION
**Pull request by s-urbaniak**
/cc @openshift/openshift-team-monitoring

### CODE REVIEW

File: manifests/client/deployment.yaml [hunk 1]
```diff
@@ -74,11 +80,8 @@ spec:
         - containerPort: 8443
           name: https
         resources:
-          limits:
-            cpu: 20m
-            memory: 40Mi
           requests:
-            cpu: 10m
+            cpu: 1m
```
Comment (squat) [inline manifests/client/deployment.yaml]: is this enough?

**Comment by squat** [comment]
it looks like this PR is built on top of a very old parent commit, which is why the integration tests are failing. Plz rebase and the tests should start passing

File: manifests/client/deployment.yaml (hunk 1 above)
Comment (s-urbaniak) [reply manifests/client/deployment.yaml]: for kube-rbac-proxy, yes, i measured it on a local e2e prometheus dump

**Comment by s-urbaniak** [comment]
@squat hmm, this is based on the newest release-4.4 commit:

```
* 16bddf7b - (HEAD -> cpu-requests, s-urbaniak/cpu-requests) assets: regenerate (32 minutes ago) <Sergiusz Urbaniak>
* 225391ac - jsonnet/telemeter: set cpu requests, unset limits (33 minutes ago) <Sergiusz Urbaniak>
*   7853182e - (origin/release-4.4, bz-1794616) Merge pull request #306 from brancz/move-metrics-whitelist (7 weeks ago) <OpenShift Merge
```

**Comment by squat** [comment]
yes, doesn't matter :) the dependency used in integration tests in ALL branches moved and needs to be fixed. You need this PR: https://github.com/openshift/telemeter/pull/317

**Comment by s-urbaniak** [comment]
ok, cherry-picked #317 commits, let's see how this goes.

**Comment by brancz** [comment]
looks good to me but having @squat review the cherry-pick

/lgtm
/hold

**Comment by squat** [comment]
To keep the backport as minimal as possible, let's drop commit bf34f2c266aa2746f1de92c5bc74d91d7f494dca, it's not needed

**Comment by s-urbaniak** [comment]
@squat fixed, PTAL

File: manifests/client/deployment.yaml (hunk 1 above)
Comment (squat) [reply manifests/client/deployment.yaml]: ack, looks like this was set as the request for every container in the pod. Even thought it's just a request, I don't think it's enough for telemeter client

**Comment by s-urbaniak** [comment]
@squat telemeter-client container peaks at 0.3 millicores
image

This is data taken from a e2e test run which we take as the reference.

ptal

**Comment by squat** [comment]
excellent :)

**Comment by s-urbaniak** [comment]
@squat or anybody out there, can i have a `/lgtm` please? :)

**Comment by s-urbaniak** [comment]
setting cherry-pick approval as this is a jsonnet library change only which will be vendored into cluster-monitoring-operator.

**Pull request merged**